### PR TITLE
create add cmocka module from scratch

### PR DESCRIPTION
**Pull request by vhvb1989**
Creating a cmake module that creates the cmocka tests exe binaries.
Removing the old cmake module that we were using as a modified version from 3rd party

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]

File: cmake-modules/AddCmockaFunctions.cmake [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (ahsonkhan) [inline cmake-modules/AddCmockaFunctions.cmake:1]: We should borrow from https://gitlab.com/cmocka/cmocka/-/blob/master/cmake/Modules/AddCMockaTest.cmake#L1 and modify as needed.

**Review by ahsonkhan — CHANGES_REQUESTED** [review]
Why did we have to write our own?

**Comment by antkmsft** [comment]
Can you find if it is possible to pass compilation flags / CMAKE_BUILD_TYPE to cmocka, at the time cmocka itself gets built? This would remove the need to suppress libcmtd warning.

**Review by antkmsft — APPROVED** [review]
As long as it does what we need, and is not very complicated, I'm not against the the idea of having our own.

File: cmake-modules/AddCmockaFunctions.cmake (hunk 1 above)
Comment (vhvb1989) [reply cmake-modules/AddCmockaFunctions.cmake:1]: I created one from scratch, with our specific needs (private access for tests to az_private, linker flags).  So we should support and maintain our own version.
We don't support wine.

**Comment by antkmsft** [comment]
If this PR goes after https://github.com/Azure/azure-sdk-for-c/pull/1338, please make sure to NOT include the libcmtd warning suppression workaround.

**Comment by vhvb1989** [comment]
> 
> 
> If this PR goes after #1338, please make sure to NOT include the libcmtd warning suppression workaround.

Yes @antkmsft , actually I have removed the warning fix and now this became blocker by https://github.com/Azure/azure-sdk-for-c/pull/1338

So, I will wait for you to merge

**Comment by antkmsft** [comment]
> > If this PR goes after #1338, please make sure to NOT include the libcmtd warning suppression workaround.
> 
> Yes @antkmsft , actually I have removed the warning fix and now this became blocker by #1338
> 
> So, I will wait for you to merge

Thank you @vhvb1989 , my PR is merged.

**Pull request closed without merging**